dist/mapdc.js

```diff
@@ -48641,6 +48646,31 @@ function rasterChart(parent, useMap, chartGroup, _mapboxgl) {
     return _legend;
   };
 
+  _chart.deleteLayerLegend = function (currentLayer, deleteLayer, prevLayerId) {
+    var legend = document.getElementsByClassName('legend');
+    if (legend.length && currentLayer !== "master") {
+      // the case that has 3 or more layers will be handled in getLegendStateFromChart function in stacked-legend.js by removing extra legend
+      if (prevLayerId && prevLayerId === "master") {
```

Comment:
Could this be simplified to prevent using an empty return and to merge the 2 `legend[deleteLayer].remove();` cases? Maybe something like:

```
if (prevLayerId && prevLayerId === "master" ||
  (!(currentLayer !== deleteLayer && prevLayerId < deleteLayer) && legend[deleteLayer])) {
  legend[deleteLayer].remove();
} else if (deleteLayer && deleteLayer !== 'master' && legend.length === 1 && legend[0]) {
  legend[0].remove();
}
```

Reply:
simplified the statement almost as same as you suggested @biovisualize  ;)